src/crimson/os/seastore/cache.cc

```diff
@@ -92,6 +97,21 @@ void Cache::retire_extent(CachedExtentRef ref)
   }
 }
 
+void Cache::replace_extent(CachedExtentRef next, CachedExtentRef prev)
+{
+  assert(next->get_paddr() == prev->get_paddr());
+  assert(next->version == prev->version + 1);
+  extents.replace(*next, *prev);
+
+  if (prev->is_dirty()) {
+    ceph_assert(prev->primary_ref_list_hook.is_linked());
+    dirty.insert(dirty.iterator_to(*prev), *next);
+    dirty.erase(dirty.iterator_to(*prev));
```

Comment:
nit, could keep an instance of `dirty.iterator_to(*prev)`, as the iterator is not invalidated after insertion, like
```c++
auto prev_it = dirty.iterator_to(*prev);
dirty.insert(prev_it, *next);
dirty.erase(prev_it);
```

for saving the overhead of the call and for creating the temporary object.